server/src/main/java/org/elasticsearch/action/admin/indices/alias/get/GetAliasesResponse.java

```diff
@@ -75,5 +126,158 @@ public void writeTo(StreamOutput out) throws IOException {
                 aliasMetaData.writeTo(out);
             }
         }
+        if (out.getVersion().onOrAfter(Version.V_7_0_0_alpha1)) {
+            // if (out.getVersion().onOrAfter(Version.V_6_3_0)) {
+            if (status != RestStatus.OK) {
+                out.writeBoolean(true);
+                out.writeInt(status.getStatus());
+                out.writeString(errorMsg);
+            } else {
+                out.writeBoolean(false);
```

Comment:
The boolean flag signals if a status and a string are following.
In this case we can always serialized the status and the error message as an optional String. The advantage of using the boolean flag is that new fields can be easily appended to the response.
Basically if in a later version the response needs to be extended with another optional String, it would be virtually impossible to distinguish if the String is 'errorMessage' or the new String field.